main.py

```diff
@@ -1,6 +1,43 @@
 """
 Codebreaker
 """
+import cProfile
+def dr_test(x, y, box, code, old_guesses, count):
+    for e in range(x):
+        guess_code = []
+        hint = []
+        print('Pick from R, B, Y, G, O, or P')
+        print('Type quit to end early')
+        for i in range(y):
+            guess = (input(f'Guess the {i + 1} position: ')).upper()
+            if guess == 'QUIT':
+                return print("You ended early, better luck next time")
+            while guess not in set(box):
+                print('Invalid input. Guess must be from R, B, Y, G, O or P')
+                guess = (input(f'Guess the {i + 1} position: ')).upper()
+            guess_code.append(guess)
+        guess_copy = guess_code.copy()
+        code_copy = code.copy()
+        for i, e in enumerate(guess_copy):
```

Comment:
I've re-done this for loop to prevent repeating code, also saw that `hint.sort()` was causing the returned "hint" to not be correct

Reply:
you'll see I've condensed the 2 for loops into 1, leveraging `if` `elseif` and `else`

also added the '-' for me personally so I could see how many indexs were being returned